lib/http/request.rb

```diff
@@ -196,6 +200,19 @@ def inspect
       "#<#{self.class}/#{@version} #{verb.to_s.upcase} #{uri}>"
     end
 
+    # @return [HTTP::URI] URI with all components but query being normalized.
+    def self.normalize_uri(uri)
```

Comment:
There's no need in duplicating default normalizer - just assign `URI::NORMALIZER` constant with lambda somethign like:

``` ruby
module HTTP
  class URI < Addressable::URI
    NORMALIZER = lambda do |uri|
      HTTP::URI.new(
        :scheme     => uri.normalized_scheme,
        # ...
      )
    end